packages/flutter/test/cupertino/switch_test.dart

```diff
@@ -513,4 +513,54 @@ void main() {
     expect(tester.widget<Opacity>(find.byType(Opacity).first).opacity, 1.0);
   });
 
+  testWidgets('Switch renders correctly before and after being tapped', (WidgetTester tester) async {
+    final Key switchKey = UniqueKey();
+    bool value = false;
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return Center(
+              child: RepaintBoundary(
+                  child: CupertinoSwitch(
+                  key: switchKey,
+                  value: value,
+                  dragStartBehavior: DragStartBehavior.down,
+                  onChanged: (bool newValue) {
+                    setState(() {
+                      value = newValue;
+                    });
+                  },
+                  )
+              )
+            );
+          },
+        ),
+      ),
+    );
+
+    await expectLater(
+      find.byKey(switchKey),
+      matchesGoldenFile(
+        'switch.tap.off.png',
+        version: 0,
+      ),
+      skip: !isLinux,
+    );
+
```

Comment:
Could you also verify that 100 milliseconds (which is half of the full duration of the animation. Or whatever timing you prefer) into the animation, the switch still has the correct appearance? Something like this:
```
    await tester.tap(find.byKey(switchKey));
   // Kick off animation.
    await tester.pump()
    await tester.pump(const Duration(milliseconds: 100));
    await expectLater(
      find.byKey(switchKey),
      matchesGoldenFile(
        'switch.tap.turningOn.png',
        version: 0,
      ),
      skip: !isLinux,
    );
```

Reply:
This is super cool. Thanks for sharing that snippet!